nameko/testing/pytest.py

```diff
@@ -84,30 +84,51 @@ def rabbit_manager(request):
 
 @pytest.yield_fixture()
 def rabbit_config(request, rabbit_manager):
+    import random
+    import string
     from kombu import pools
-    from nameko.testing.utils import (
-        reset_rabbit_vhost, reset_rabbit_connections,
-        get_rabbit_connections, get_rabbit_config)
+    from nameko.testing.utils import get_rabbit_connections
+    from six.moves.urllib.parse import urlparse  # pylint: disable=E0611
 
     amqp_uri = request.config.getoption('AMQP_URI')
 
-    conf = get_rabbit_config(amqp_uri)
+    uri = urlparse(amqp_uri)
+    username = uri.username
+    vhost = uri.path[1:]
 
-    reset_rabbit_connections(conf['vhost'], rabbit_manager)
-    reset_rabbit_vhost(conf['vhost'], conf['username'], rabbit_manager)
+    use_random_vost = vhost == ":random:"
+
+    if use_random_vost:
+        vhost = "test_{}".format(
+            "".join(random.choice(string.lowercase) for _ in range(10))
+        )
+        amqp_uri = amqp_uri.replace(":random:", vhost)
```

Comment:
above you parse the url to explicitly compare the value for vhost to `':random:'`, but here your run the replace over the whole string. seems inconsistent.